DigitalLearningSolutions.Data.Tests/Services/UserVerificationServiceTests.cs

```diff
@@ -0,0 +1,340 @@
+        public void
+            GetVerifiedAdminUserAssociatedWithDelegateUsers_Returns_nothing_when_no_admin_account_is_associated_with_delegate()
+        {
+            // Given
+            var delegateUser = UserTestHelper.GetDefaultDelegateUser();
+            var delegateUsers = new List<DelegateUser> { delegateUser };
+            A.CallTo(() => userDataService.GetAdminUserByEmailAddress(A<string>._)).Returns(null);
+
+            // When
+            var returnedAdminUser = userVerificationService.GetVerifiedAdminUserAssociatedWithDelegateUsers(
+                delegateUsers,
+                "password"
+            );
+
+            // Then
+            Assert.IsNull(returnedAdminUser);
+        }
+
+        [Test]
+        public void
+            GetVerifiedAdminUserAssociatedWithDelegateUsers_Returns_nothing_when_admin_account_found_is_at_different_centre()
+        {
+            // Given
+            var delegateUser = UserTestHelper.GetDefaultDelegateUser(centreId: 2);
+            var delegateUsers = new List<DelegateUser> { delegateUser };
+            var associatedAdminUser = UserTestHelper.GetDefaultAdminUser(centreId: 5);
+            A.CallTo(() => userDataService.GetAdminUserByEmailAddress(delegateUser.EmailAddress!))
+                .Returns(associatedAdminUser);
+            A.CallTo(() => cryptoService.VerifyHashedPassword(A<string>._, A<string>._)).Returns(true);
+
+            // When
+            var returnedAdminUser = userVerificationService.GetVerifiedAdminUserAssociatedWithDelegateUsers(
+                delegateUsers,
+                "password"
+            );
+
+            // Then
+            Assert.IsNull(returnedAdminUser);
+        }
+
+        [Test]
+        public void
+            GetVerifiedAdminUserAssociatedWithDelegateUser_Returns_verified_admin_account_associated_with_delegate_by_email()
+        {
+            // Given
+            var delegateUser = UserTestHelper.GetDefaultDelegateUser();
+            var delegateUsers = new List<DelegateUser> { delegateUser };
+            var associatedAdminUser = UserTestHelper.GetDefaultAdminUser();
+            A.CallTo(() => userDataService.GetAdminUserByEmailAddress(delegateUser.EmailAddress!))
+                .Returns(associatedAdminUser);
+            A.CallTo(() => cryptoService.VerifyHashedPassword(A<string>._, A<string>._)).Returns(true);
+
+            // When
+            var returnedAdminUser = userVerificationService.GetVerifiedAdminUserAssociatedWithDelegateUsers(
+                delegateUsers,
+                "password"
+            );
+
+            // Then
+            Assert.AreEqual(associatedAdminUser, returnedAdminUser);
+        }
+
+        [Test]
+        public void GetVerifiedDelegateUsersAssociatedWithAdminUser_Returns_empty_list_when_admin_is_null()
+        {
+            // When
+            var returnedDelegates = userVerificationService.GetVerifiedDelegateUsersAssociatedWithAdminUser(
+                null,

```

Comment:
Not a big deal, but in cases like this, I like having the test input value set in a `given` section - that makes it clear that the special case you're testing is where `adminUser = null` rather than, say, where `password = "password"`.